### PR TITLE
[uservm] CLI Arguments Support for Hyperlight

### PR DESCRIPTION
**Pull request by ppenna**
This pull request significantly improves the way the Hyperlight platform passes and handles initrd (initial ramdisk) arguments from the user VM host to the guest kernel. The changes introduce a robust, length-prefixed encoding for initrd command-line arguments, ensure correct relocation and validation of the initrd image, and add comprehensive error checking throughout the process. Additionally, the tests are updated to verify the correct passing of the executable name as argv[0].

**Initrd argument handling and encoding:**

* Added a new method `build_args_bytes` in `src/uservm/src/vmm/hyperlight/mod.rs` to encode the program name and arguments as a length-prefixed byte vector, ensuring the guest receives argv[0] and any arguments correctly. This method also enforces a maximum length of 255 bytes for arguments.
* The user VM now appends the encoded arguments to the initrd blob passed to the guest, and adjusts memory size calculations to account for the header and arguments.

**Initrd parsing and relocation in the guest kernel:**

* Refactored the guest kernel's `parse_bootinfo` in `src/kernel/src/hal/platform/hyperlight/mod.rs` to use a new `parse_initrd_image` function. This function reads the initrd header, validates sizes, relocates the initrd if necessary, and extracts the argument section. [[1]](diffhunk://#diff-43171bb5ac9a97f5483ae1e974f64ec7c70221d6c6c657f874cab96e16b3827eR242-L316) [[2]](diffhunk://#diff-43171bb5ac9a97f5483ae1e974f64ec7c70221d6c6c657f874cab96e16b3827eR407-R606)
* Added `read_initrd_cmdline`, which robustly reads and validates the length-prefixed command-line arguments from the initrd payload, ensuring correct UTF-8 encoding and handling edge cases.

**Testing improvements:**

* Updated the test in `src/tests/misc-c/main.c` to assert that `argv[0]` is set to the executable name, verifying the correctness of the argument passing mechanism.

**Code quality and safety:**

* Improved error handling and logging throughout the initrd parsing and argument handling code, with clear error messages for allocation, overflow, and encoding issues. [[1]](diffhunk://#diff-43171bb5ac9a97f5483ae1e974f64ec7c70221d6c6c657f874cab96e16b3827eR407-R606) [[2]](diffhunk://#diff-2eeb0b110bec3866ba7a09cd14837a8dc2f0142ea909f4da49d666a2fad0b9c1L132-R181)
* Refactored imports and removed unused code for clarity and maintainability. [[1]](diffhunk://#diff-43171bb5ac9a97f5483ae1e974f64ec7c70221d6c6c657f874cab96e16b3827eR35-R45) [[2]](diffhunk://#diff-43171bb5ac9a97f5483ae1e974f64ec7c70221d6c6c657f874cab96e16b3827eL60-L63) [[3]](diffhunk://#diff-2eeb0b110bec3866ba7a09cd14837a8dc2f0142ea909f4da49d666a2fad0b9c1R18) [[4]](diffhunk://#diff-2eeb0b110bec3866ba7a09cd14837a8dc2f0142ea909f4da49d666a2fad0b9c1R40)

These changes collectively make the initrd and argument passing mechanism more robust, secure, and easier to maintain.

### CODE REVIEW

File: src/kernel/src/hal/platform/hyperlight/mod.rs [hunk 1]
```diff
@@ -445,3 +404,203 @@ pub fn init(
...
+    let args_bytes_offset: usize = match args_len_offset.checked_add(1) {
+        Some(offset) => offset,
+        None => {
+            let reason: &str = "initrd arguments payload address overflow";
+            error!("read_initrd_cmdline(): {reason}");
+            return Err(Error::new(ErrorCode::BadFile, reason));
+        },
+    };
+
+    // Check if arguments length byte is missing>
```
[the remark this answers is not in the document]
Comment (ppenna) [reply src/kernel/src/hal/platform/hyperlight/mod.rs]: Fixed.

File: src/uservm/src/vmm/hyperlight/mod.rs [hunk 2]
```diff
@@ -375,4 +391,51 @@ impl Vmm {
...
+    ///
+    /// - `program_name`: The name of the program to be executed.
+    /// - `program_args`: An optional string containing the arguments to be passed to the program.
+    ///
+    /// # Returns
+    ///
+    /// On success, this function returns a vector of bytes representing the encoded arguments.
+    /// On failure, it returns an error.
+    ///
+    fn build_args_bytes(program_name: &String, program_args: &Option<String>) -> Result<Vec<u8>> {
```
[the remark this answers is not in the document]
Comment (ppenna) [reply src/uservm/src/vmm/hyperlight/mod.rs:412]: This is ok.
